### ISSUE TITLE
CAST(0x AS BINARY)

### ISSUE THREAD

**Issue by fleming42** [phpmyadmin/phpmyadmin#16773]
Found a bug where phpmyadmin 5.1.0 constructs an invalid query when selecting "edit" on a row with an empty string in a BLOB or TEXT column. The resultant query has "CAST(0x AS BINARY)" in it, but 0x by itself is invalid. Looks like phpmyadmin expects a nonempty $row to be appended to the '0x' in Util.php.

My fix for this is in Util.php, by changing the function getConditionValue from:
```php
            // do not waste memory building a too big condition
            if (mb_strlen((string) $row) < 1000) {
                // use a CAST if possible, to avoid problems
                // if the field contains wildcard characters % or _
```
to
```php
            // do not waste memory building a too big condition
            if ((mb_strlen((string) $row) > 0)
                && (mb_strlen((string) $row) < 1000)
            ) {
                // use a CAST if possible, to avoid problems
                // if the field contains wildcard characters % or _
```

Works for me now, but please examine this and see if this is right. Thanks.

**Comment by williamdes** [comment]
Hi @fleming42 
Thank you for the nice detailed report, I will have a look
To improve your fix for performance ;)
```php
            $rowLength = mb_strlen((string) $row);
            if (($rowLength > 0) && ($rowLength < 1000)) {
```

**Comment by williamdes** [comment]
Could you confirm @fleming42 that this is the test case you are talking about ?

![Sélection_2021040102:02:32001](https://user-images.githubusercontent.com/7784660/113225625-60ae9e80-928e-11eb-9cc5-3157c6e13fc3.jpg)

```sql
-- phpMyAdmin SQL Dump
-- version 5.1.1-dev
-- https://www.phpmyadmin.net/
--
-- Hôte : mysql803.williamdes.local:3306
-- Généré le : jeu. 01 avr. 2021 à 00:03
-- Version du serveur : 8.0.3-rc-log
-- Version de PHP : 7.4.14

SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
START TRANSACTION;
SET time_zone = "+00:00";

--
-- Base de données : `public`
--

-- --------------------------------------------------------

--
-- Structure de la table `textx`
--

CREATE TABLE `textx` (
  `tex` text CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

--
-- Déchargement des données de la table `textx`
--

INSERT INTO `textx` (`tex`) VALUES
('0'),
('');

--
-- Index pour les tables déchargées
--

--
-- Index pour la table `textx`
--
ALTER TABLE `textx`
  ADD UNIQUE KEY `U` (`tex`(253));
COMMIT;
```

**Comment by williamdes** [comment]
I made the fix a12706165594868ab348c2be97ea710e1caf0bb8, you can apply it if you want and let me know if it works :)

**Comment by fleming42** [comment]
Yes, your test case is the same as mine, and yes fix a127061 works for me.
Thank you. :-)

David Fleming
www.flemingcomputer.com
